### ISSUE TITLE
Amplify pull causes an issue

### ISSUE THREAD

**Issue by Abhishek01039** [aws-amplify/amplify-cli#8233]
### Before opening, please confirm:

- [X] I have installed the latest version of the Amplify CLI (see above), and confirmed that the issue still persists.
- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-cli/issues?q=is%3Aissue+).
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-cli/blob/master/CONTRIBUTING.md#bug-reports).
- [X] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

### How did you install the Amplify CLI?

npm

### If applicable, what version of Node.js are you using?

v15.9.0

### Amplify CLI Version

5.3.1

### What operating system are you using?

Mac

### Amplify Categories

auth, storage, api, hosting

### Amplify Commands

pull

### Describe the bug

Whenever I am trying to pull the existing backend with amplify I got this error
```
Backend environment dev found in Amplify Console app: console_app
Failed to pull the backend.
Cannot read property 'Body' of null
TypeError: Cannot read property 'Body' of null
    at downloadBackend (/Users/abhishek/.nvm/versions/node/v15.9.0/lib/node_modules/@aws-amplify/cli/node_modules/amplify-provider-awscloudformation/src/attach-backend.js:327:38)
    at processTicksAndRejections (node:internal/process/task_queues:94:5)
    at Object.run (/Users/abhishek/.nvm/versions/node/v15.9.0/lib/node_modules/@aws-amplify/cli/node_modules/amplify-provider-awscloudformation/src/attach-backend.js:77:3)
    at Object.attachBackend (/Users/abhishek/.nvm/versions/node/v15.9.0/lib/node_modules/@aws-amplify/cli/node_modules/amplify-provider-awscloudformation/src/index.ts:46:3)
    at Object.queryProvider (/Users/abhishek/.nvm/versions/node/v15.9.0/lib/node_modules/@aws-amplify/cli/src/attach-backend-steps/a10-queryProvider.ts:12:3)
    at Object.attachBackend (/Users/abhishek/.nvm/versions/node/v15.9.0/lib/node_modules/@aws-amplify/cli/src/attach-backend.ts:20:5)
    at Object.run (/Users/abhishek/.nvm/versions/node/v15.9.0/lib/node_modules/@aws-amplify/cli/src/commands/pull.ts:59:5)
    at Object.executeAmplifyCommand (/Users/abhishek/.nvm/versions/node/v15.9.0/lib/node_modules/@aws-amplify/cli/src/index.ts:362:5)
    at executePluginModuleCommand (/Users/abhishek/.nvm/versions/node/v15.9.0/lib/node_modules/@aws-amplify/cli/src/execution-manager.ts:177:3)
    at Object.executeCommand (/Users/abhishek/.nvm/versions/node/v15.9.0/lib/node_modules/@aws-amplify/cli/src/execution-manager.ts:29:5)
    at Object.run (/Users/abhishek/.nvm/versions/node/v15.9.0/lib/node_modules/@aws-amplify/cli/src/index.ts:200:5)
```

### Expected behavior

It's should not cause an issue and pull the backend from amplify

### Reproduction steps

1. Run ` amplify pull --appId app_id --envName dev`
2. Enter accessKeyId
3. Enter secretAccessKey
4. select region

### GraphQL schema(s)

<details>

```graphql
# Put schemas below this line


```

</details>

### Log output

<details>

```
# Put your logs below this line


```

</details>

**Comment by attilah** [comment]
@Abhishek01039 could you attach the logs from `~/.amplify/logs` ? also verify that you have the zip file for your backend in your S3 Deployment bucket?

Is the same error happening if you'd use an AWS profile for authentication?

**Comment by josefaidt** [comment]
Hey @Abhishek01039 I'm unable to reproduce this issue using the steps provided. Can you provide the logs as @attilah mentioned? That should provide more insight into the root cause.

**Comment by Abhishek01039** [comment]
@attilah and @josefaidt 
Thanks for seeing this issue

this is logs
```
2021-09-21T14:50:10.574Z|info : amplify pull core  {"appId":"[***]cny3f3","envName":"[***]ev","yes":false}
2021-09-21T14:50:10.631Z|info : amplify-cli-core.banner-message/index.ts.fetch banner messages from https://aws-amplify.github.io/amplify-cli/banner-message.json({}
2021-09-21T14:50:32.901Z|info : amplify-provider-awscloudformation.amplify-service-permission-check.checkAmplifyServiceIAMPermission.amplifyClient.listApps([])
2021-09-21T14:50:34.010Z|info : amplify-provider-awscloudformation.attach-backend.getAmplifyApp.amplifyClient.getApp([{"appId":"[***]cny3f3"}])
2021-09-21T14:50:34.936Z|info : amplify-provider-awscloudformation.attach-backend.getBackendEnv.amplifyClient.getBackendEnvironment([{"appId":"[***]cny3f3","environmentName":"[***]ev"}])
2021-09-21T14:50:35.815Z|info : amplify-provider-awscloudformation.attach-backend.downloadBackend.s3.getObject([{"Key":"[***]rent-[***]ud-[***]d.zip","Bucket":"[***]ify-[***]k92-[***]ev-[***]650-[***]ment"}])
2021-09-21T14:50:36.980Z|error : amplify-provider-awscloudformation.attach-backend.downloadBackend.s3.getObject([{"Key":"[***]rent-[***]ud-[***]d.zip","Bucket":"[***]ify-[***]k92-[***]ev-[***]650-[***]ment"}])
NoSuchBucket: The specified bucket does not exist
2021-09-21T15:13:35.674Z|info : amplify configure core  
2021-09-21T15:13:35.759Z|info : amplify-cli-core.banner-message/index.ts.fetch banner messages from https://aws-amplify.github.io/amplify-cli/banner-message.json({}
2021-09-21T16:12:20.173Z|info : amplify pull core  {"appId":"[***]cny3f3","envName":"[***]ev","yes":false}
2021-09-21T16:12:20.243Z|info : amplify-cli-core.banner-message/index.ts.fetch banner messages from https://aws-amplify.github.io/amplify-cli/banner-message.json({}
2021-09-21T16:12:46.997Z|info : amplify-provider-awscloudformation.amplify-service-permission-check.checkAmplifyServiceIAMPermission.amplifyClient.listApps([])
2021-09-21T16:12:47.912Z|info : amplify-provider-awscloudformation.attach-backend.getAmplifyApp.amplifyClient.getApp([{"appId":"[***]cny3f3"}])
2021-09-21T16:12:48.715Z|info : amplify-provider-awscloudformation.attach-backend.getBackendEnv.amplifyClient.getBackendEnvironment([{"appId":"[***]cny3f3","environmentName":"[***]ev"}])
2021-09-21T16:12:49.482Z|info : amplify-provider-awscloudformation.attach-backend.downloadBackend.s3.getObject([{"Key":"[***]rent-[***]ud-[***]d.zip","Bucket":"[***]ify-[***]k92-[***]ev-[***]650-[***]ment"}])
2021-09-21T16:12:50.607Z|error : amplify-provider-awscloudformation.attach-backend.downloadBackend.s3.getObject([{"Key":"[***]rent-[***]ud-[***]d.zip","Bucket":"[***]ify-[***]k92-[***]ev-[***]650-[***]ment"}])
NoSuchBucket: The specified bucket does not exist
2021-09-21T16:15:06.689Z|info : amplify version core  
```

**Comment by josefaidt** [comment]
Hey @Abhishek01039 thanks for posting those logs. I was able to reproduce the issue by manually deleting the s3 deployment bucket for an existing project. Can you confirm whether the deployment bucket for this project is available in the s3 console?

**Comment by attilah** [comment]
@Abhishek01039 that error you are seeing is originated from the fact that the deployment bucket or the objects that Amplify CLI manages there were deleted. Debugging the issue led to a fix that we were missing that is fixed in #8233 now you would see a clear error about what was missing instead of the reported error.

**Comment by Abhishek01039** [comment]
Sure thanks @attilah and @josefaidt
